dotnet/targets/Xamarin.Shared.Sdk.targets

```
@@ -2416,6 +2416,41 @@ global using nfloat = global::System.Runtime.InteropServices.NFloat%3B
 		</ItemGroup>
 	</Target>
 
+	<Target Name="ComputeXcsyncArguments">
+		<PropertyGroup>
+			<xcSyncExecute>$(DOTNET_HOST_PATH) $(XcsyncPath)</xcSyncExecute>
+			<xcSyncProjectFile Condition="'$(xcSyncProjectFile)' == ''">$(MSBuildProjectDirectory)/$(MSBuildProjectFile)</xcSyncProjectFile>
+			<xcSyncXcodeFolder Condition="'$(xcSyncXcodeFolder)' == ''">$(IntermediateOutputPath)xcode</xcSyncXcodeFolder>
+			<xcSyncTargetFrameworkMoniker Condition="'$(xcSyncTargetFrameworkMoniker)' == ''">$(TargetFramework)</xcSyncTargetFrameworkMoniker>
+			<xcSyncVerbosity Condition="'$(xcSyncVerbosity)' == ''">Diagnostic</xcSyncVerbosity>
+
+
+			<XcsyncCommonArguments Condition="'$(xcSyncProjectFile)' != ''">$(XcsyncCommonArguments) -p $(xcSyncProjectFile)</XcsyncCommonArguments>
```

Quote to handle project directories with spaces:

```suggestion
			<XcsyncCommonArguments Condition="'$(xcSyncProjectFile)' != ''">$(XcsyncCommonArguments) -p "$(xcSyncProjectFile)"</XcsyncCommonArguments>
```